stdlib/public/Distributed/DistributedActorSystem.swift

```diff
@@ -249,12 +243,37 @@ extension DistributedActorSystem {
     }
 
     do {
+      // Decode the invocation and pack arguments into the h-buffer
+      // TODO(distributed): decode the generics info
+      var argumentDecoder = invocation.makeArgumentDecoder()
+      var paramIdx = 0
+      for unsafeRawArgPointer in hargs {
+        guard paramIdx < paramCount else {
+          throw ExecuteDistributedTargetError(
+            message: "Unexpected attempt to decode more parameters than expected: \(paramIdx + 1)")
+        }
+        let paramType = paramTypes[paramIdx]
+        paramIdx += 1
+
+        // FIXME(distributed): func doDecode<Arg: SerializationRequirement>(_: Arg.Type) throws {
+        // FIXME:     but how would we call this...?
+        // FIXME:     > type 'Arg' constrained to non-protocol, non-class type 'Self.Invocation.SerializationRequirement'
+        func doDecodeArgument<Arg>(_: Arg.Type) throws {
+          let unsafeArgPointer = unsafeRawArgPointer
+            .bindMemory(to: Arg.self, capacity: 1)
+          try argumentDecoder.decodeNext(Arg.self, into: unsafeArgPointer)
+        }
+        try _openExistential(paramType, do: doDecodeArgument)
```

Comment:
I think this is okay for now but in the future we probably want to encapsulate this into `HeterogeneousBuffer` since it has all the necessary type information already, via a method like `HeterogeneousBuffer::populate(from: ArgumentDecoder)`.

Reply:
Right, okey for now and as discussed today we'll have to stuff it into _executeDistributedTarget eventually.